controllers/controllers.go

```diff
@@ -106,6 +106,31 @@ func (ctrl *OAuthController) HandleTokenRequest(w http.ResponseWriter, r *http.R
 	return ctrl.token(w, ctrl.Service.OauthServer.GetTokenData(ti), nil)
 }
 
+func (ctrl *OAuthController) TokenIntrospectHandler(w http.ResponseWriter, r *http.Request) {
+	token := strings.TrimPrefix(r.Header.Get("Authorization"), "Bearer ")
+	tokenInfo, err := ctrl.Service.OauthServer.Manager.LoadAccessToken(r.Context(), token)
+	if err != nil {
+		logrus.Errorf("Something went wrong loading access token: %s, token %s, request %v", err.Error(), token, r)
+		sentry.CaptureException(err)
+	}
+	// for middleware
+	lti := r.Context().Value("token_info")
+	if lti != nil {
+			logTokenInfo := lti.(*models.LogTokenInfo)
+			logTokenInfo.UserId = tokenInfo.GetUserID()
+			logTokenInfo.ClientId = tokenInfo.GetClientID()
+	}
+	scopes := map[string]string{}
```

Comment:
I would return a bit more structured response with some more information like the redirect URI and the client id.
```
{
"client_id":"",
"redirect_uri":"",
"scopes": {..}
}
```
So you will need to return a map[string]interface{}. I would also include documentation for this endpoint in the README.